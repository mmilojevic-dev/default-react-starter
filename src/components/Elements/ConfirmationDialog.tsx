import { ExclamationTriangleIcon, InfoCircledIcon } from '@radix-ui/react-icons'
import * as React from 'react'

import {
  Button,
  Dialog,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
  DialogTrigger
} from '@/components'
import { useOpenClose } from '@/hooks/useOpenClose'

export type ConfirmationDialogProps = {
  triggerButton: React.ReactElement
  confirmButton: React.ReactElement
  title: string
  body?: string
  cancelButtonText?: string
  icon?: 'danger' | 'info'
  isDone?: boolean
}

export const ConfirmationDialog = ({
  triggerButton,
  confirmButton,
  title,
  body = '',
  cancelButtonText = 'Cancel',
  icon = 'danger',
  isDone = false
}: ConfirmationDialogProps) => {
  const { isOpen, close, toggle } = useOpenClose()

  React.useEffect(() => {
    if (isDone) {
      close()
    }
  }, [isDone, close])

  return (
    <Dialog open={isOpen} onOpenChange={toggle}>
      <DialogTrigger asChild>{triggerButton}</DialogTrigger>
      <DialogContent>
        <DialogHeader className="gap-y-4">
          <div className="flex items-center gap-x-2">
            {icon === 'danger' && (
              <ExclamationTriangleIcon className="size-6 text-destructive" />
            )}
            {icon === 'info' && (
              <InfoCircledIcon className="size-6 text-accent" />
            )}
            <DialogTitle>{title}</DialogTitle>
          </div>
        </DialogHeader>
        <DialogDescription className="px-8">{body}</DialogDescription>
        <DialogFooter>
          <div className="mt-4 flex space-x-2">
            <Button variant="outline" onClick={close}>
              {cancelButtonText}
            </Button>
            {confirmButton}
          </div>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  )
}
