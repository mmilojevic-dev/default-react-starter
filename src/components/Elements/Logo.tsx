import { Target } from 'lucide-react'
import { Link } from 'react-router-dom'

import { APP_DEFAULT } from '@/config'
import { cn } from '@/utils'

interface LogoProps {
  withTitle?: boolean
  size?: number
}

export const Logo = ({ withTitle = false, size = 8 }: LogoProps) => {
  const sizeClass = `size-${size}`
  return (
    <Link className="flex items-center gap-2" to=".">
      <Target className={cn(sizeClass, 'rounded-full', 'text-primary')} />
      {withTitle && (
        <span className="text-xs font-semibold">{APP_DEFAULT.TITLE}</span>
      )}
    </Link>
  )
}
