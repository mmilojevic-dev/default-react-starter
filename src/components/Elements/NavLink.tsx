import { NavLink as RouterNavLink, useLocation } from 'react-router-dom'

import { ROUTES } from '@/config'
import { cn } from '@/utils'

type ProtectedRouteKey = keyof typeof ROUTES.PROTECTED

type NavLinkProps = {
  route: (typeof ROUTES.PROTECTED)[ProtectedRouteKey]
}

export const NavLink = ({ route }: NavLinkProps) => {
  const { pathname } = useLocation()

  const isActive = route.PATH === pathname
  const linkClasses = cn(
    isActive ? 'bg-background/50 text-foreground' : 'text-muted-foreground/75',
    `group flex uppercase tracking-widest items-center rounded-md p-2 font-medium hover:text-foreground`
  )
  const iconClasses = cn(
    isActive ? 'text-foreground' : 'text-muted-foreground/75',
    'mr-4 size-6 shrink-0 group-hover:text-foreground'
  )

  return (
    <RouterNavLink className={linkClasses} to={route.PATH}>
      {route.ICON && <route.ICON className={iconClasses} aria-hidden="true" />}
      {route.LABEL}
    </RouterNavLink>
  )
}
