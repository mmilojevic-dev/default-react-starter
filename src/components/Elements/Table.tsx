import { ArchiveIcon } from 'lucide-react'
import React from 'react'

export type TableColumnType<Entry> = {
  title: string
  field: keyof Entry
  Cell?: ({ entry }: { entry: Entry }) => React.ReactNode
}

export type TableProps<Entry> = {
  data: Entry[]
  columns: TableColumnType<Entry>[]
}

export const Table = <Entry extends object>({
  data,
  columns
}: TableProps<Entry>) => {
  if (!data?.length) {
    return (
      <div className="flex h-80 flex-col items-center justify-center bg-background text-muted">
        <ArchiveIcon className="size-16" />
        <h4>No Entries Found</h4>
      </div>
    )
  }

  return (
    <div className="max-h-[65vh] w-full overflow-auto">
      <table className="min-w-full divide-y divide-border text-sm">
        <thead className="sticky top-0 z-10 bg-background text-foreground">
          <tr>
            {columns.map((col) => (
              <th
                key={col.title}
                className="text-nowrap p-3 text-left text-xs font-medium uppercase tracking-wider"
              >
                {col.title}
              </th>
            ))}
          </tr>
        </thead>
        <tbody className="divide-y divide-border">
          {data.map((entry, index) => (
            <tr key={index}>
              {columns.map((col) => (
                <td
                  key={`${String(col.field)}-${index}`}
                  className="px-3 py-4 text-sm font-medium"
                >
                  <span>
                    {col.Cell ? col.Cell({ entry }) : String(entry[col.field])}
                  </span>
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  )
}
