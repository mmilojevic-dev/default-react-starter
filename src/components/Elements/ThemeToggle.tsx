import { Moon, Sun } from 'lucide-react'
import React from 'react'
import { useDispatch } from 'react-redux'

import {
  Button,
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuTrigger
} from '@/components'
import { THEME } from '@/config'
import { AppDispatch, setTheme } from '@/store'
import { ThemeEnum } from '@/types'

export const ThemeToggle: React.FC = () => {
  const dispatch = useDispatch<AppDispatch>()

  return (
    <DropdownMenu>
      <DropdownMenuTrigger asChild>
        <Button variant="outline" size="icon">
          <span className="sr-only">Toggle theme</span>
          <Sun className="size-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0" />
          <Moon
            className="absolute size-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0
              dark:scale-100"
          />
          <span className="sr-only">{THEME.TOGGLER_LABEL}</span>
        </Button>
      </DropdownMenuTrigger>
      <DropdownMenuContent align="end">
        <DropdownMenuItem
          className="capitalize"
          onClick={() => dispatch(setTheme(ThemeEnum.Light))}
        >
          {ThemeEnum.Light}
        </DropdownMenuItem>
        <DropdownMenuItem
          className="capitalize"
          onClick={() => dispatch(setTheme(ThemeEnum.Dark))}
        >
          {ThemeEnum.Dark}
        </DropdownMenuItem>
        <DropdownMenuItem
          className="capitalize"
          onClick={() => dispatch(setTheme(ThemeEnum.System))}
        >
          {ThemeEnum.System}
        </DropdownMenuItem>
      </DropdownMenuContent>
    </DropdownMenu>
  )
}
