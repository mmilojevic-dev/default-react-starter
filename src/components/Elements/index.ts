export * from './Button'
export * from './ConfirmationDialog'
export * from './Dialog'
export * from './Drawer'
export * from './DropdownMenu'
export * from './Logo'
export * from './NavLink'
export * from './Spinner'
export * from './Table'
