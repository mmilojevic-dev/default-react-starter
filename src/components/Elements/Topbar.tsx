import { MenuIcon } from 'lucide-react'

import { Button } from '@/components/Elements/Button'
import { ThemeToggle } from '@/components/Theme/ThemeToggle'

type TopbarProps = {
  setMobileDrawerOpen: React.Dispatch<React.SetStateAction<boolean>>
}

export const Topbar = ({ setMobileDrawerOpen }: TopbarProps) => {
  return (
    <div className="bg-popover/15 relative z-10 flex h-16 shrink-0">
      <div className="flex flex-1 justify-between px-4">
        <div className="mr-4 flex items-center md:mr-6">
          <Button
            variant="outline"
            size="icon"
            className="md:hidden"
            onClick={() => setMobileDrawerOpen(true)}
          >
            <span className="sr-only">Open sidebar</span>
            <MenuIcon className="size-6" aria-hidden="true" />
          </Button>
        </div>
        <div className="ml-4 flex items-center md:ml-6">
          <ThemeToggle />
        </div>
      </div>
    </div>
  )
}
