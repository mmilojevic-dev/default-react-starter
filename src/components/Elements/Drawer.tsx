import * as DrawerRadix from '@radix-ui/react-dialog'
import { Cross2Icon } from '@radix-ui/react-icons'
import { cva, type VariantProps } from 'class-variance-authority'
import * as React from 'react'

import { Button } from '@/components'
import { cn } from '@/utils'

const Drawer = DrawerRadix.Root

const DrawerTrigger = DrawerRadix.Trigger

const DrawerClose = DrawerRadix.Close

const DrawerPortal = DrawerRadix.Portal

const DrawerOverlay = React.forwardRef<
  React.ElementRef<typeof DrawerRadix.Overlay>,
  React.ComponentPropsWithoutRef<typeof DrawerRadix.Overlay>
>(({ className, ...props }, ref) => (
  <DrawerRadix.Overlay
    className={cn(
      `fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in
      data-[state=closed]:animate-out data-[state=closed]:fade-out-0
      data-[state=open]:fade-in-0`,
      className
    )}
    {...props}
    ref={ref}
  />
))
DrawerOverlay.displayName = DrawerRadix.Overlay.displayName

const DrawerVariants = cva(
  'fixed z-50 min-h-0 w-screen gap-4 overflow-y-auto bg-popover shadow-lg transition ease-in-out data-[state=closed]:duration-300 data-[state=open]:duration-500 data-[state=open]:animate-in data-[state=closed]:animate-out',
  {
    variants: {
      side: {
        top: 'inset-x-0 top-0 border-b data-[state=closed]:slide-out-to-top data-[state=open]:slide-in-from-top',
        bottom:
          'inset-x-0 bottom-0 border-t data-[state=closed]:slide-out-to-bottom data-[state=open]:slide-in-from-bottom',
        left: 'inset-y-0 left-0 h-full w-3/4 border-r data-[state=closed]:slide-out-to-left data-[state=open]:slide-in-from-left sm:max-w-sm',
        right:
          'inset-y-0 right-0 h-full w-3/4 border-l data-[state=closed]:slide-out-to-right data-[state=open]:slide-in-from-right sm:max-w-sm'
      },
      size: {
        xs: 'max-w-60',
        sm: 'max-w-xs',
        md: 'max-w-md',
        lg: 'max-w-4xl',
        full: 'max-w-full'
      }
    },
    defaultVariants: {
      side: 'left',
      size: 'xs'
    }
  }
)

interface DrawerContentProps
  extends React.ComponentPropsWithoutRef<typeof DrawerRadix.Content>,
    VariantProps<typeof DrawerVariants> {}

const DrawerContent = React.forwardRef<
  React.ElementRef<typeof DrawerRadix.Content>,
  DrawerContentProps
>(({ side = 'left', className, children, ...props }, ref) => (
  <DrawerPortal>
    <DrawerOverlay />
    <DrawerRadix.Content
      ref={ref}
      className={cn(DrawerVariants({ side }), className)}
      {...props}
    >
      {children}
      <DrawerRadix.Close className="absolute right-3 top-3 disabled:pointer-events-none">
        <Button variant="outline">
          <Cross2Icon className="size-4" />
          <span className="sr-only">Close</span>
        </Button>
      </DrawerRadix.Close>
    </DrawerRadix.Content>
  </DrawerPortal>
))
DrawerContent.displayName = DrawerRadix.Content.displayName

const DrawerHeader = ({
  className,
  ...props
}: React.HTMLAttributes<HTMLDivElement>) => (
  <div
    className={cn(
      'flex flex-col space-y-2 text-center sm:text-left',
      className
    )}
    {...props}
  />
)
DrawerHeader.displayName = 'DrawerHeader'

const DrawerBody = ({
  className,
  ...props
}: React.HTMLAttributes<HTMLDivElement>) => (
  <div className={cn('flex flex-col space-y-2', className)} {...props} />
)
DrawerBody.displayName = 'DrawerBody'

const DrawerFooter = ({
  className,
  ...props
}: React.HTMLAttributes<HTMLDivElement>) => (
  <div
    className={cn(
      'flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2',
      className
    )}
    {...props}
  />
)
DrawerFooter.displayName = 'DrawerFooter'

const DrawerTitle = React.forwardRef<
  React.ElementRef<typeof DrawerRadix.Title>,
  React.ComponentPropsWithoutRef<typeof DrawerRadix.Title>
>(({ className, ...props }, ref) => (
  <DrawerRadix.Title
    ref={ref}
    className={cn('text-lg font-semibold text-foreground', className)}
    {...props}
  />
))
DrawerTitle.displayName = DrawerRadix.Title.displayName

const DrawerDescription = React.forwardRef<
  React.ElementRef<typeof DrawerRadix.Description>,
  React.ComponentPropsWithoutRef<typeof DrawerRadix.Description>
>(({ className, ...props }, ref) => (
  <DrawerRadix.Description
    ref={ref}
    className={cn('text-sm text-muted-foreground', className)}
    {...props}
  />
))
DrawerDescription.displayName = DrawerRadix.Description.displayName

export {
  Drawer,
  DrawerBody,
  DrawerClose,
  DrawerContent,
  DrawerDescription,
  DrawerFooter,
  DrawerHeader,
  DrawerOverlay,
  DrawerPortal,
  DrawerTitle,
  DrawerTrigger
}
