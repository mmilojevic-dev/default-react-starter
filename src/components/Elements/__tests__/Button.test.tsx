import { fireEvent, render, screen } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'

import { Button, buttonVariants } from '@/components/Elements/Button'

// Explicitly defined variants and sizes
const variantKeys = [
  'default',
  'destructive',
  'outline',
  'secondary',
  'ghost',
  'link'
] as const
const sizeKeys = ['default', 'sm', 'lg', 'icon'] as const

describe('Button', () => {
  it('renders correctly with default props', () => {
    // Render the Button component with default props
    const { asFragment } = render(<Button>Click me</Button>)

    // Verify that the button is rendered with the correct text and classes
    const button = screen.getByText('Click me')
    expect(button).toBeInTheDocument()
    expect(button).toHaveClass(
      'inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2 hover:bg-primary/90 bg-primary text-primary-foreground'
    )

    // Generate snapshot
    expect(asFragment()).toMatchSnapshot()
  })

  variantKeys.forEach((variant) => {
    it(`renders with ${variant} variant`, () => {
      // Render the Button component with each variant
      const { asFragment } = render(<Button variant={variant}>Click me</Button>)

      // Verify that the button has the correct variant classes
      const button = screen.getByText('Click me')
      expect(button).toHaveClass(buttonVariants({ variant }))

      // Generate snapshot
      expect(asFragment()).toMatchSnapshot()
    })
  })

  sizeKeys.forEach((size) => {
    it(`renders with ${size} size`, () => {
      // Render the Button component with each size
      const { asFragment } = render(<Button size={size}>Click me</Button>)

      // Verify that the button has the correct size classes
      const button = screen.getByText('Click me')
      expect(button).toHaveClass(buttonVariants({ size }))

      // Generate snapshot
      expect(asFragment()).toMatchSnapshot()
    })
  })

  it('handles click event', () => {
    // Mock the click handler function
    const handleClick = vi.fn()
    const { asFragment } = render(
      <Button onClick={handleClick}>Click me</Button>
    )

    // Click the button and verify that the click handler is called
    const button = screen.getByText('Click me')
    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalledTimes(1)

    // Generate snapshot
    expect(asFragment()).toMatchSnapshot()
  })
})
