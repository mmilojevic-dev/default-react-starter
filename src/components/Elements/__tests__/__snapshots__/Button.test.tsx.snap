// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > handles click event 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-primary/90 bg-primary text-primary-foreground h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders correctly with default props 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-primary/90 bg-primary text-primary-foreground h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with default size 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-primary/90 bg-primary text-primary-foreground h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with default variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-primary/90 bg-primary text-primary-foreground h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with destructive variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-destructive/90 bg-destructive text-destructive-foreground h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with ghost variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with icon size 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-primary/90 bg-primary text-primary-foreground size-10"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with lg size 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-primary/90 bg-primary text-primary-foreground h-11 rounded-md px-8"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with link variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with outline variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with secondary variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-secondary/80 bg-secondary text-secondary-foreground h-10 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Button > renders with sm size 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-primary/90 bg-primary text-primary-foreground h-9 rounded-md px-3"
  >
    Click me
  </button>
</DocumentFragment>
`;
